### ISSUE TITLE
pip install failed on Ubuntu 22.04.2 LTS / Debian 12.0 

### ISSUE THREAD

**Issue by seanedevane** [glmcdona/binary2strings#20]
Since the 0.1.10 release I'm having installs failing when running `pip3 install`. Initial failures were on a Ubuntu 22.04.2 LTS box, with Python 3.10.6 / pip3 23.1.2 installed within a venv configuration. I've since replicated it on Debian 12 with Python 3.11.2 and Kali 2023.2 also with Python 3.11.2 / pip3 23.0.1. The build will fail both within a venv as well as with an user install with the following trace:

```sh
$ python3 -m venv test       
$ source test/bin/activate
                                                                                                                     
┌──(test)
└─$ pip3 install binary2strings
Collecting binary2strings
  Downloading binary2strings-0.1.10.tar.gz (9.3 kB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Installing backend dependencies ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: binary2strings
  Building wheel for binary2strings (pyproject.toml) ... error
  error: subprocess-exited-with-error
  
  × Building wheel for binary2strings (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      running bdist_wheel
      running build
      running build_ext
      x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -I/home/rokkr/Scripts/test-binary-strings/test/include -I/usr/include/python3.11 -c flagcheck.cpp -o flagcheck.o -std=c++17
      building 'binary2strings' extension
      creating build
      creating build/temp.linux-x86_64-cpython-311
      creating build/temp.linux-x86_64-cpython-311/src
      x86_64-linux-gnu-gcc -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -DVERSION_INFO=0.1.10 -Isrc -I/tmp/pip-build-env-fv5_42_5/overlay/lib/python3.11/site-packages/pybind11/include -I/home/rokkr/Scripts/test-binary-strings/test/include -I/usr/include/python3.11 -c src/binary2strings.cpp -o build/temp.linux-x86_64-cpython-311/src/binary2strings.o -std=c++17 -fvisibility=hidden -g0
      src/binary2strings.cpp:1:10: fatal error: binary2strings.hpp: No such file or directory
          1 | #include "binary2strings.hpp"
            |          ^~~~~~~~~~~~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for binary2strings
Failed to build binary2strings
ERROR: Could not build wheels for binary2strings, which is required to install pyproject.toml-based projects
```

**Comment by glmcdona** [comment]
Thanks, looking into it.

**Comment by seanedevane** [comment]
I had some time to do a little more digging. The root cause seems to be that the source tarball `pip install` tries to pull from is missing the .hpp files.

I tested building the release manually with `python3 setup.py sdist` and in my first test this package was then installed successfully and contained all the files needed in the /src directory. Python 3.112, setuptools 66.1.1, and pybind11 2.11.1 were on that system.

On a different VM I tried to replicate the process, but that time generating the release didn't include the .hpp files in the src. For that I had Python 3.10.6, setuptools 67.8.0, and pybind11 2.1.1. Trying to use Python3.11 on that machine still resulted in not copying over the .hpp files, so I may be missing some useful information about the configurations for you.

**Comment by glmcdona** [comment]
I suspect is it the change I made to setup.py here:
https://github.com/glmcdona/binary2strings/commit/2c812c2deda5b9540328c0b06e9d99041840b18a

I removed that line of code because a warning message indicated that `headers` was unrecognized, but maybe it actually was being used. I'll rebuild with reverting that change to see if that fixes it.

**Comment by glmcdona** [comment]
That didn't fix the issue, looking at it closer now.

**Comment by glmcdona** [comment]
Fixed. Had to add a MANIFEST.in to include the .hpp files in the distribution. ChatGPT-4 was a lifesaver here, pointed out the issue right away correctly. Tested on Ubuntu 18.04.5 LTS and Windows 11.

Published fix with version 0.1.13.

**Comment by seanedevane** [comment]
Confirming that fixed it in my builds as well, thanks!
